conversion/conversion.go

```diff
@@ -328,16 +299,33 @@ func dataFromCSV(ctx context.Context, sourceProfile profiles.SourceProfile, targ
 	if err != nil {
 		return nil, err
 	}
+
 	totalRows := conv.Rows()
 	p := internal.NewProgress(totalRows, "Writing data to Spanner", internal.Verbose(), false)
+	batchWriter := populateDataConv(conv, config, client, p)
+	err = csv.ProcessCSV(conv, tables, sourceProfile.Csv.NullStr, delimiter)
+	if err != nil {
+		return nil, fmt.Errorf("can't process csv: %v", err)
+	}
+	batchWriter.Flush()
+	p.Done()
+	return batchWriter, nil
+}
+
+func populateDataConv(conv *internal.Conv, config writer.BatchWriterConfig, client *sp.Client, progress *internal.Progress) *writer.BatchWriter {
 	rows := int64(0)
 	config.Write = func(m []*sp.Mutation) error {
-		_, err := client.Apply(context.Background(), m)
+		migrationData := migration.MigrationData{
+			MigrationRequestId: &conv.MigrationRequestId,
+		}
```

Comment:
Replace with migrationData := metrics.GetMigrationData(...). Let GetMigrationData decide what needs to be populated in migration data for all use cases.